### PR TITLE
Review gathering

### PR DESCRIPTION
**Pull request by andrewalbrecht05**
### Project Description💡
This Telegram bot simplifies graph visualization for users, particularly in competitive programming contexts. It empowers users to seamlessly convert textual graph representations into clear and aesthetically pleasing visualizations.

### Key Features⚙️
Effortless Graph Input: Users submit a list of edges (and optional labels) representing their graph structure.
Customizable Output: The bot generates an image of the graph, employing the Graphviz library to render the visuals.
Directed/Undirected Support: Handles both directed and undirected graphs.
Intuitive Interaction: Guides users through the visualization process with clear commands and prompts.

### CODE REVIEW

**Comment by Dagunov** [comment]
Cool bot, simple code and design!  
Special thanks for readme with examples!  

As a feature request for future updates (if planned) - maybe consider allowing to export graphs in `.drawio` format, allowing to open them on [draw.io](https://app.diagrams.net/)!

**Review by Dagunov — APPROVED** [review]

**Review by imlvts — APPROVED** [review]
Looks good. I don't know if this is the best UI for this task, but it certainly solves the problem.

**Review by heckq — APPROVED** [review]
I like to solve algo tasks on leetcode so i find this bot quite helpful
For the future, consider adding features that show graph’s shortest path(djkstra algorithm)

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+use regex::Regex;
+use unicode_segmentation::UnicodeSegmentation;
+
+/// Represents a single node within a graph
+#[derive(Debug)]
+struct Node {
+    label: String,
+}
+impl Node {
+    pub fn new(label: String) -> Self {
```
Comment (divi255) [inline src/lib.rs]: нащо тут pub якщо сама структура приватна?

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+/// Represents a connection (edge) between two nodes in a graph
+#[derive(Debug)]
+struct Edge {
+    from: String,
+    to: String,
+    label: String,
+}
+impl Edge {
+    pub fn new(from: String, to: String, label: String) -> Self {
```
Comment (divi255) [inline src/lib.rs]: то саме

**Review by divi255 — APPROVED** [review]

File: src/handlers.rs [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    let direction_choice = msg.text().unwrap().trim();
+    trace!("This is message in receive state: {}", direction_choice);
+
+    // Parses the user's input to determine if the graph is directed
+    let directed: Option<bool> = match direction_choice.to_lowercase().as_str() {
+        "y" => Some(true),
+        "n" => Some(false),
+        _ => None
```
Comment (IvanSraichuk) [inline src/handlers.rs]: Can't you just return here and use `directed: bool`?

**Review by IvanSraichuk — APPROVED** [review]

File: src/handlers.rs (hunk 3 above)
Comment (andrewalbrecht05) [reply src/handlers.rs]: I agree, this would be better
fixed

**Pull request merged**